include/pdal/Metadata.hpp

```diff
@@ -560,7 +560,7 @@ class PDAL_DLL MetadataNode
     }
 
     bool hasChildren() const
-        { return m_impl->m_subnodes.size(); }
+        { return (m_impl->m_subnodes.size()) != 0; }
```

Comment:
I guess I'm old.  The casting of ints to bools seems superfluous.  If you don't know that this is part of the language, then it seems like you don't know the language.  I think someone just got assigned some task like "Make the compiler emit lots of warnings."

I'm done now :)